File: dummy_creator.php
<?php

namespace user_reg_protocol;


// base include
require_once __DIR__.'/../generic_protocol/dummy_creator.php';
// includes
require_once __DIR__.'/../basic_parser/dummy_creator.php';
require_once __DIR__.'/../basic_objects/dummy_creator.php';

// own includes
require_once __DIR__.'/../user_reg_protocol/object_initializer.php';

// enums

// objects

function create_dummy__User()
{
    $res = new User;

    initialize__User( $res
        , \basic_objects\create_dummy__gender_e()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_parser\create_dummy__string()
        , \basic_objects\create_dummy__Email()
        , \basic_parser\create_dummy__string()
        , \basic_objects\create_dummy__Date()
        );
    return $res;
}

// messages

function create_dummy__RegisterUserRequest()
{
    $res = new RegisterUserRequest;

    initialize__RegisterUserRequest( $res
        , create_dummy__User()
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

function create_dummy__RegisterUserResponse()
{
    $res = new RegisterUserResponse;

    initialize__RegisterUserResponse( $res
        );
    return $res;
}

function create_dummy__ConfirmRegistrationRequest()
{
    $res = new ConfirmRegistrationRequest;

    initialize__ConfirmRegistrationRequest( $res
        , \basic_parser\create_dummy__string()
        );
    return $res;
}

function create_dummy__ConfirmRegistrationResponse()
{
    $res = new ConfirmRegistrationResponse;

    initialize__ConfirmRegistrationResponse( $res
        );
    return $res;
}

// namespace_end user_reg_protocol


?>

File: dummy_creator.cpp
// includes
#include "dummy_creator.h"
#include "generic_protocol/dummy_creator.h"
#include "basic_parser/dummy_creator.h"
#include "basic_objects/dummy_creator.h"
#include "object_initializer.h"

namespace user_reg_protocol
{

namespace dummy
{

// enums

// objects

User create__User()
{
    User res;

    ::user_reg_protocol::initialize( & res
        , ::basic_objects::dummy::create__gender_e()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_parser::dummy::create__string()
        , ::basic_objects::dummy::create__Email()
        , ::basic_parser::dummy::create__string()
        , ::basic_objects::dummy::create__Date()
        );

    return res;
}

// messages

RegisterUserRequest * create__RegisterUserRequest()
{
    auto res = new RegisterUserRequest;

    ::user_reg_protocol::initialize( res
        , dummy::create__User()
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

RegisterUserResponse * create__RegisterUserResponse()
{
    auto res = new RegisterUserResponse;

    ::user_reg_protocol::initialize( res
        );

    return res;
}

ConfirmRegistrationRequest * create__ConfirmRegistrationRequest()
{
    auto res = new ConfirmRegistrationRequest;

    ::user_reg_protocol::initialize( res
        , ::basic_parser::dummy::create__string()
        );

    return res;
}

ConfirmRegistrationResponse * create__ConfirmRegistrationResponse()
{
    auto res = new ConfirmRegistrationResponse;

    ::user_reg_protocol::initialize( res
        );

    return res;
}

} // namespace dummy

} // namespace user_reg_protocol

